### ISSUE TITLE
OpenMPI builds unrequested/unwelcome memchecker version for some apps

### ISSUE THREAD

**Issue by dslarm** [LLNL/spack#36255]
### Steps to reproduce

When building some packages (but not every package), adding MKL or ArmPL to the spec causes +memchecker to be added to openmpi spec yielding a slow performance in the final build
```console
spack spec hpcg%gcc@12.2.0 ^openmpi ^intel-oneapi-mkl
spack spec hpcg%gcc@12.2.0 ^openmpi ^armpl-gcc
```
Similar happens for wrf, but not for gromacs.

```
$ spack spec hpcg%gcc@12.2.0 ^openmpi ^intel-oneapi-mkl
Input spec
--------------------------------
hpcg%gcc@12.2.0
    ^intel-oneapi-mkl
    ^openmpi

Concretized
--------------------------------
hpcg@3.1%gcc@12.2.0+openmp build_system=autotools patches=722c138 arch=linux-amzn2-icelake
    ^openmpi@4.1.5%gcc@12.2.0~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~java~legacylaunchers~lustre+memchecker~orterunprefix+romio+rsh~singularity+static+vt+wrapper-rpath build_system=autotools fabrics=none schedulers=none arch=linux-amzn2-icelake
```

### Information on your system

* **Spack:** 0.20.0.dev0 (5f8ee20c7c58ece53d8df8f996ae059331567c46)
* **Python:** 3.7.16
* **Platform:** linux-amzn2-icelake
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
We could make the `memchecker` variant a sticky variant, if @hppritcha and @naughtont3 agree. For reference:

https://spack.readthedocs.io/en/latest/packaging_guide.html#sticky-variants

**Comment by hppritcha** [comment]
sticky attribute sounds interesting.

**Comment by alalazo** [comment]
Fixed by #36329
